janus_server/src/datastore.rs

```diff
@@ -859,20 +858,29 @@ impl Transaction<'_> {
             )
             .await?;
 
-        // Found an existing collect job
-        if let Some(row) = row {
-            return Ok(row.get("collect_job_id"));
-        }
+        Ok(row.map(|row| row.get("collect_job_id")))
```

Comment:
Q: It's sorta strange that this returns an `Option` determined by whether the row exists -- all other methods return the found object directly, and if there is no such object, they return `Err(NotFound)`.

An `Option` is the more Rust-idiomatic way to return an optional value. OTOH, all other methods "expect" the object to exist, so it's an error-worthy surprising condition if they don't -- that doesn't apply here, I suppose.

I'm going to make this uniform across all methods soon; I was wondering if you had any thoughts on which way (`Option` vs `NotFound`) would be better.

Reply:
I'm inclined toward `Option` in cases like this where we may expect one or zero rows. I just wrote a match block that pulled out `Error::NotFound` on `get_task()`, and I can see us having to do this more as we implement all the anti-replay logic.

Reply:
I prefer using an `Option` for the same reasons as David: if we use `Err()` only for errors, then we can always just propagate those with `?`.